### PR TITLE
Integration with ODIM's services part #6

### PR DESCRIPTION
**Pull request by MaciejMis**
Changes:
- Implemented GET action for all resources under System resource.

### CODE REVIEW

File: src/rest/server.go [hunk 1]
```diff
@@ -33,6 +33,49 @@ func createRouting(app *iris.Application, config config.Config) {
...
+		systems.Get("/{id}/Bios", getGenericResourceHandler)
+		systems.Get("/{id}/Bios/Settings", getGenericResourceHandler)
+
+		ethernetInterfaces := systems.Party("/{id}/EthernetInterfaces")
+		ethernetInterfaces.Get("", getGenericResourceHandler)
+		ethernetInterfaces.Get("/{id}", getGenericResourceHandler)
+		ethernetInterfaces.Get("/{id}/VLANS", getGenericResourceHandler)
+		ethernetInterfaces.Get("/{id}/VLANS/{id2}", getGenericResourceHandler)
+
+		logService := systems.Party("{id}/LogServices")
```
Comment (ehaligow) [inline src/rest/server.go:56]: Isn't here a slash missing?

File: src/rest/server.go [hunk 2]
```diff
@@ -33,6 +33,49 @@ func createRouting(app *iris.Application, config config.Config) {
...
+		storage.Get("/{id}/Volumes/{id2}", getGenericResourceHandler)
+		storage.Get("/{id}/Drives/{id2}", getGenericResourceHandler)
+
+		storagePools := storage.Party("/{id}/StoragePools")
+		storagePools.Get("", getGenericResourceHandler)
+		storagePools.Get("/{id}", getGenericResourceHandler)
+		storagePools.Get("/{id}/AllocatedVolumes", getGenericResourceHandler)
+		storagePools.Get("/{id}/AllocatedVolumes/{id2}", getGenericResourceHandler)
+
+		capacitySources := storagePools.Party("{id}/CapacitySources")
```
Comment (ehaligow) [inline src/rest/server.go:75]: Isn't here a slash missing?

File: src/rest/server.go [hunk 3]
```diff
@@ -33,6 +33,49 @@ func createRouting(app *iris.Application, config config.Config) {
...
+		ethernetInterfaces.Get("/{id}/VLANS", getGenericResourceHandler)
+		ethernetInterfaces.Get("/{id}/VLANS/{id2}", getGenericResourceHandler)
+
+		logService := systems.Party("{id}/LogServices")
+		logService.Get("", getGenericResourceHandler)
+		logService.Get("/{id}", getGenericResourceHandler)
+		logService.Get("/{id}/Entries", getGenericResourceHandler)
+		logService.Get("/{id}/Entries/{id2}", getGenericResourceHandler)
+
+		storage := routes.Party("/{id}/Storage")
```
[the remark this answers is not in the document]
Comment (jcleung5549) [reply src/rest/server.go:62]: Per the schema, the Storage resource can be subordinate to Systems, ServiceRoot, etc.

```
        <Annotation Term="Redfish.Uris">
          <Collection>
            <String>/redfish/v1/Storage/{StorageId}</String>
            <String>/redfish/v1/Systems/{ComputerSystemId}/Storage/{StorageId}</String>
            <String>/redfish/v1/CompositionService/ResourceBlocks/{ResourceBlockId}/Storage/{StorageId}</String>
            <String>/redfish/v1/CompositionService/ResourceBlocks/{ResourceBlockId}/Systems/{ComputerSystemId}/Storage/{StorageId}</String>
            <String>/redfish/v1/ResourceBlocks/{ResourceBlockId}/Storage/{StorageId}</String>
            <String>/redfish/v1/ResourceBlocks/{ResourceBlockId}/Systems/{ComputerSystemId}/Storage/{StorageId}</String>
          </Collection>
        </Annotation>
```

[the remark this answers is not in the document]
Comment (ehaligow) [reply src/rest/server.go:62]: I see, thanks for clarification.

File: src/rest/server.go (hunk 1 above)
Comment (MaciejMis) [reply src/rest/server.go:56]: Fixed

File: src/rest/server.go (hunk 2 above)
Comment (MaciejMis) [reply src/rest/server.go:75]: Fixed

File: src/rest/server.go [hunk 4]
```diff
@@ -33,6 +33,49 @@ func createRouting(app *iris.Application, config config.Config) {
...
+		systems.Get("/{id}/BootOptions", getGenericResourceHandler)
+		systems.Get("/{id}/BootOptions/{id2}", getGenericResourceHandler)
+		systems.Get("/{id}/Processors", getGenericResourceHandler)
+		systems.Get("/{id}/Processors/{id2}", getGenericResourceHandler)
+		systems.Get("/{id}/Memory", getGenericResourceHandler)
+		systems.Get("/{id}/Memory/{id2}", getGenericResourceHandler)
+		systems.Get("/{id}/NetworkInterfaces", getGenericResourceHandler)
+		systems.Get("/{id}/MemoryDomains", getGenericResourceHandler)
+		systems.Get("/{id}/SecureBoot", getGenericResourceHandler)
+		systems.Get("/{id}/NetworkInterfaces/{id2}", getGenericResourceHandler)
```
[the remark this answers is not in the document]
Comment (MaciejMis) [reply src/rest/server.go:45]: I moved endpoints to different groups only when the path had 3+ different endpoints.

**Comment by ehaligow** [comment]
Looks good.

**Pull request merged**